### PR TITLE
DOC: add examples for index related functions

### PR DESCRIPTION
**Pull request by hagenw**
This adds the following examples to the docstring of index related functions:

### filewise_index

image

### segmented_index

image

### index_type

image

### CODE REVIEW

File: audformat/core/index.py [hunk 1]
```diff
@@ -103,6 +113,23 @@ def segmented_index(
     Raises:
         ValueError: if ``files``, ``start`` and ``ends`` differ in size
 
+    Example:
+        >>> segmented_index(['a.wav', 'b.wav'])
+        MultiIndex([('a.wav', '0 days', NaT),
+                    ('b.wav', '0 days', NaT)],
+                   names=['file', 'start', 'end'])
+        >>> segmented_index(['a.wav'], [0], [1])
```
Comment (frankenjoe) [inline audformat/core/index.py:121]:
```suggestion
        >>> segmented_index('a.wav', '0s', '1s')
```

File: audformat/core/index.py [hunk 2]
```diff
@@ -103,6 +113,23 @@ def segmented_index(
...
+        MultiIndex([('a.wav', '0 days', '0 days 00:00:00.000000001')],
+                   names=['file', 'start', 'end'])
+        >>> segmented_index(
+        ...     files=['a.wav', 'a.wav'],
+        ...     starts=[t * 10 ** 9 for t in (0, 1)],
```
Comment (frankenjoe) [inline audformat/core/index.py:126]: I find this syntax a little bit too complicated for a usage example.

File: audformat/core/index.py [hunk 3]
```diff
@@ -103,6 +113,23 @@ def segmented_index(
...
+        ...     ends=[pd.Timedelta(t, unit='s') for t in (1, 2)],
```
Comment (frankenjoe) [inline audformat/core/index.py:127]:
```suggestion
        ...     ends=pd.to_timedelta([1, 2], unit='s'),
```

**Comment by frankenjoe** [comment]
We should may also also include an example for `segmented_index` where `starts` and `ends' is not set, and maybe one where only one of it is set.

File: audformat/core/index.py (hunk 2 above)
Comment (hagenw) [reply audformat/core/index.py:126]: Would the following be better:
```python
starts=[f'{t}s' for t in (0, 1)],
```
For me the above line is still easier then using `pd.Timedelta` as I can never remember how to use that one.
The idea behind this line is also to make it easier when https://github.com/audeering/audformat/issues/12 is addressed.

Comment (frankenjoe) [reply audformat/core/index.py:126]: What about just `['0s', '1s']`?

Comment (frankenjoe) [reply audformat/core/index.py:126]: or `['0s', '1000ms']` to show that other units are supported, too?

Comment (hagenw) [reply audformat/core/index.py:126]: Yes `['0s', '1s']` it's fine for an example, but in reality you most probably have already a long list with integers or floats and not strings.

Comment (frankenjoe) [reply audformat/core/index.py:126]: Ok, let's use `[f'{t}s' for t in (0, 1)]` here and for `ends` use `pd.to_timedelta([1, 2], unit='s')`. Then we show two scalable options.

File: audformat/core/index.py (hunk 1 above)
Comment (hagenw) [reply audformat/core/index.py]: I updated this.

File: audformat/core/index.py (hunk 2 above)
Comment (hagenw) [reply audformat/core/index.py]: Changed it accordingly.

**Comment by hagenw** [comment]
> We should may also also include an example for `segmented_index` where `starts` and `ends` are not set, and maybe one where only one of it is set.

The first was already there, for the other two I added one additional example.

**Pull request merged**